src/constraints/StringConstraints.ts

```diff
@@ -44,3 +57,68 @@ export function stringLengthNe(length: number): IConstraint<string> {
 	const expected = `expected.length !== ${length}`;
 	return stringLengthComparator(ne, 's.string.lengthNe', expected, length);
 }
+
+export function stringRegex(regex: RegExp, type: 'url' | 'uuid' | 'regex' | 'email'): IConstraint<string> {
+	const expected = `expected.${type}`;
```

Comment:
```suggestion
	const expected = `expected to be a ${type}`;
```

Reply:
I just realised, this has a little small flaw.

- `expected to be a url` (should be `to match an URL`)
- `expected to be a uuid` (should be `to match an UUID`)
- `expected to be a regex` (???????, should be `to match ${pattern}`)
- `expected to be a email` (should be `to match an email`)

Reply:
updated